components/FacetMenu/FacetMenu.vue

```diff
@@ -0,0 +1,235 @@
+          <span class="label">{node.label}</span>
+        </span>
+      )
+    },
+    // If the facet is a top level facet then it is one of the categories and cannot be selected,
+    // We can only hide the checkbox so it still shows up as selected when all its children are checked
+    isTopLevelNode(node) {
+      return node.level <= 1
+    },
+    onFacetChecked(clickedNode, treeStatus) {
+      this.selectedFacets = this.$refs.tree
+        .getCheckedNodes()
+        .filter(checkedNode => {
+          const node = this.$refs.tree.getNode(checkedNode)
+          return (
+            !this.isTopLevelNode(node) &&
+            (node.parent.indeterminate || this.isTopLevelNode(node.parent))
+          )
+        })
+      this.$emit('selected-facets-changed', this.selectedFacets)
+    },
+    deselectAllFacets() {
+      this.facets.forEach(facet => {
+        this.$refs.tree.setChecked(facet.id, false, true)
+      })
+      this.selectedFacets = []
+      this.$emit('selected-facets-changed', this.selectedFacets)
+    },
+    deselectAllChildrenFacets(node) {
+      node.children.forEach(child => {
+        this.$refs.tree.setChecked(child.id, false, true)
+        this.selectedFacets = this.$refs.tree.getCheckedNodes()
+      })
+      this.$emit('selected-facets-changed', this.selectedFacets)
+    },
+    deselectFacet(id) {
+      this.$refs.tree.setChecked(id, false, true)
+      this.selectedFacets = this.selectedFacets.filter(facet => facet.id != id)
+      this.$emit('selected-facets-changed', this.selectedFacets)
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+@import '../../assets/_variables.scss';
+
+.facets {
+  background: white;
+
+  h2 {
+    font-size: 1.25rem;
+    font-weight: 500;
+    line-height: 1.2;
+  }
+
+  .title {
+    margin-bottom: 0;
+    padding: 0.5rem 1rem;
+  }
+
+  hr {
+    border: none;
+    border-bottom: 1px solid #dbdfe6;
+    margin: 0;
+  }
+
+  .el-tree-node {
+    border-bottom: 1px solid #dbdfe6;
+  }
+
+  .el-tree-node__children > .el-tree-node {
+    border-bottom: none;
+  }
+
+  .el-tree > .el-tree-node > .el-tree-node__content {
+    text-transform: uppercase;
+    label.el-checkbox {
+      display: none;
+    }
+  }
+
+  .el-tree > .el-tree-node {
+    padding: 0.5rem 1rem;
+  }
+
+  .el-tree-node__content {
+    display: flex;
+    .el-tree-node__expand-icon {
+      order: 3;
+      margin-left: auto;
+    }
+  }
+
+  .el-tree-node__children {
+    padding: 0 10px;
+  }
+
+  .el-checkbox__input.is-checked .el-checkbox__inner,
+  .el-checkbox__input.is-indeterminate .el-checkbox__inner {
+    border-color: $median !important;
+    background-color: $median !important;
+  }
+  .el-checkbox__input.is-checked + .el-checkbox__label {
+    color: $median !important;
+  }
+  .el-checkbox__inner:hover {
+    border-color: $median !important;
+  }
+  .el-checkbox__input.is-focus .el-checkbox__inner {
+    border-color: $median !important;
+  }
+
+  .custom-tree-node {
```

Comment:
Agreed. I am going to focus on those details once I have implemented the actual faceting logic. Basically I may have hamstrung myself by utilizing the el-tree component because it is not as customizable as I anticipated since little things like changing the background color for sections and the show less/show more functionality are proving to be very tricky

Reply:
I used the el-cascader (not entirely sure I would recommend that one either but it does work) when making some search filters on the /maps page. I know what you mean hahaha